cdap-app-templates/cdap-etl/cdap-etl-core/src/main/java/io/cdap/cdap/etl/lineage/FieldLineageProcessor.java

```diff
@@ -70,6 +70,12 @@ public FieldLineageProcessor(PipelineSpec pipelineSpec) {
     for (StageSpec stageSpec : pipelineSpec.getStages()) {
 
       Map<String, Schema> inputSchemas = stageSpec.getInputSchemas();
+      // workaround to avoid NPE if the schema is not set
+      // TODO: CDAP-16428 populate the schema if macro is enabled to avoid this
+      if (inputSchemas == null || inputSchemas.isEmpty()) {
```

Comment:
It looks like we also assume all map values are non-null. That is not correct and should also be guarded against.